python/pacemaker/_cts/validate.py

```diff
@@ -0,0 +1,61 @@
+"""A module providing XML validation utilities."""
+
+__copyright__ = "Copyright 2024 the Pacemaker project contributors"
+__license__ = "GNU General Public License version 2 or later (GPLv2+)"
+
+__all__ = ["validate"]
+
+import os
+import subprocess
+import sys
+
+from pacemaker._cts.errors import XmlValidationError
+from pacemaker._cts.process import pipe_communicate
+from pacemaker.buildoptions import BuildOptions
+
+
+def find_validator(rng_file):
+    """
+    Return the command line used to validate XML output.
+
+    If no validator is found, return None.
+    """
+    if os.access(BuildOptions.XMLLINT_PATH, os.X_OK):
+        if rng_file is None:
+            return ["xmllint", "-"]
+
+        return [BuildOptions.XMLLINT_PATH, "--relaxng", rng_file, "-"]
+
+    raise FileNotFoundError("Could not find validator for %s" % rng_file)
+
+
+def rng_directory():
+    """Return the directory containing RNG schema files."""
+    if "PCMK_schema_directory" in os.environ:
+        return os.environ["PCMK_schema_directory"]
+
+    if os.path.exists("%s/cts-fencing.in" % sys.path[0]):
```

Comment:
This is the python module search path, not $PATH (which I had to double check as well), and we can ensure it will be first based on the sys.path.insert calls at the top of cts-cli.in.  However, I can change it to search through the path instead of grabbing the first thing.

Also, this is a weird check in the first place.  I'm not sure where I came up with it.  We should probably either search some other environment variable or look for a different file.

Reply:
I think the check was copy-and-paste from elsewhere. It's fine though, all that matters is we're picking a file that's unique to pacemaker and only in a checkout (.in).